packages/flutter_tools/gradle/aar_init_script.gradle

```diff
@@ -0,0 +1,120 @@
+            }
+        }
+    }
+    Boolean usesFlutterPlugin = project.plugins.find { it.class.name == "FlutterPlugin" } != null
+    if (!usesFlutterPlugin) {
+        // Plugins don't include their dependencies under the assumption that the parent project adds them.
+        if (project.properties['android.useAndroidX']) {
+            project.dependencies {
+                compileOnly "androidx.annotation:annotation:+"
+            }
+        } else {
+            project.dependencies {
+                compileOnly "com.android.support:support-annotations:+"
+            }
+        }
+        project.dependencies {
+            // The Flutter plugin already adds `flutter.jar`.
+            compileOnly project.files("${getFlutterRoot(project)}/bin/cache/artifacts/engine/android-arm-release/flutter.jar")
+        }
+    }
+}
+
+String getFlutterRoot(Project project) {
+    if (!project.hasProperty("flutter-root")) {
+        throw new GradleException("The `-Pflutter-root` flag must be specified.")
+    }
+    return project.property("flutter-root")
+}
+
+void addAarTask(Project project, variant) {
+    String variantName = variant.name.capitalize()
+    String taskName = "assembleAar${variantName}"
+    project.tasks.create(name: taskName) {
+        // This check is required to be able to configure the archives before `uploadArchives` runs.
+        if (!project.gradle.startParameter.taskNames.contains(taskName)) {
+            return
+        }
+        // NOTE(blasten): `android.defaultPublishConfig` must equal the variant name to build.
+        // Where variant name is `<product-flavor><Build-Type>`. However, it's too late to configure
+        // `defaultPublishConfig` at this point. Therefore, the code below ensures that the
+        // default build config uses the artifacts produced for the specific build variant.
+        Task bundle = project.tasks.findByName("bundle${variantName}Aar") // gradle:3.2.0
+        if (bundle == null) {
+            bundle = project.tasks.findByName("bundle${variantName}") // gradle:3.1.0
+        }
+        if (bundle == null) {
+            throw new GradleException("Can't generate AAR for variant ${variantName}.");
+        }
+        project.uploadArchives.repositories.mavenDeployer {
+            pom {
+                artifactId = "${project.name}_${variant.name.toLowerCase()}"
+            }
+        }
+        // Clear the current archives since the current one is assigned based on
+        // `android.defaultPublishConfig` which defaults to `release`.
+        project.configurations["archives"].artifacts.clear()
+        // Add the artifact that will be published.
+        project.artifacts.add("archives", bundle)
+        // Generate the Maven artifacts.
+        finalizedBy "uploadArchives"
+    }
+}
+
+projectsEvaluated {
+    // Plugin projects don't have subprojects.
+    // TODO(blasten): Have a more robust way of detecting plugins.
```

Comment:
What if a plugin does have subprojects? Does this break?

Is there some way we can just check if com.android.library plugin is applied?

Reply:
This is a flag now.